streams/src/main/java/org/apache/kafka/streams/kstream/internals/StreamStreamJoinUtil.java

```diff
@@ -25,36 +25,46 @@
 
 public final class StreamStreamJoinUtil {
 
-    private StreamStreamJoinUtil(){
+    private StreamStreamJoinUtil() {
     }
 
     public static <KIn, VIn, KOut, VOut> boolean skipRecord(
         final Record<KIn, VIn> record, final Logger logger,
         final Sensor droppedRecordsSensor,
-        final ProcessorContext<KOut, VOut> context) {
+        final ProcessorContext<KOut, VOut> context
+    ) {
         // we do join iff keys are equal, thus, if key is null we cannot join and just ignore the record
         //
         // we also ignore the record if value is null, because in a key-value data model a null-value indicates
         // an empty message (ie, there is nothing to be joined) -- this contrast SQL NULL semantics
         // furthermore, on left/outer joins 'null' in ValueJoiner#apply() indicates a missing record --
         // thus, to be consistent and to avoid ambiguous null semantics, null values are ignored
         if (record.key() == null || record.value() == null) {
-            if (context.recordMetadata().isPresent()) {
-                final RecordMetadata recordMetadata = context.recordMetadata().get();
-                logger.warn(
-                    "Skipping record due to null key or value. "
-                        + "topic=[{}] partition=[{}] offset=[{}]",
-                    recordMetadata.topic(), recordMetadata.partition(), recordMetadata.offset()
-                );
-            } else {
-                logger.warn(
-                    "Skipping record due to null key or value. Topic, partition, and offset not known."
-                );
-            }
-            droppedRecordsSensor.record();
+            logSkip("null key or value", logger, droppedRecordsSensor, context);
             return true;
         } else {
             return false;
         }
     }
+
+    public static <KOut, VOut> void logSkip(
+        final String reason,
+        final Logger logger,
+        final Sensor droppedRecordsSensor,
+        final ProcessorContext<KOut, VOut> context
+    ) {
+        if (context.recordMetadata().isPresent()) {
+            final RecordMetadata recordMetadata = context.recordMetadata().get();
+            logger.warn(
+                "Skipping record. reason=[{}] topic=[{}] partition=[{}] offset=[{}]",
```

Comment:
Nit: Just comparing to `AbstractKStreamTimeWindowAggregateProcessor#logSkippedRecordForExpiredWindow` is seems we could addd more information -- should we try to merge both "skip reason" as your PR proposes, if have two different output, one key null-case, and a different one for "expired" case similar to windowed-aggregation?